src/env.cr

```diff
@@ -52,6 +52,29 @@ module ENV
     !!LibC.getenv(key)
   end
 
+  # Fetch an environment variable corresponding to a key. Throws an exception
+  # if the key does not exist. If a second argument or a block is provided then
+  # this value is returned instead of throwing the exception.
+  #
+  def self.fetch(key)
+    fetch(key) do
+      if block = @@block
+        block.call(self, key)
+      else
+        raise KeyError.new "Missing ENV key: #{key.inspect}"
+      end
+    end
+  end
+
+  def self.fetch(key, default)
+    fetch(key) { default }
+  end
+
+  def self.fetch(key)
+    return self[key] if has_key?(key)
+    yield(key).to_s
+  end
```

Comment:
Also reimplementing `[]`, and `[]?` in terms of this one should DRY up the class a little.